api/features/feature_external_resources/models.py

```diff
@@ -0,0 +1,95 @@
+import logging
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from django_lifecycle import (
+    AFTER_SAVE,
+    BEFORE_DELETE,
+    LifecycleModelMixin,
+    hook,
+)
+
+from features.models import Feature, FeatureState
+from integrations.github.github import generate_data
+from integrations.github.tasks import call_github_app_webhook_for_feature_state
+from webhooks.webhooks import WebhookEventType
+
+logger = logging.getLogger(__name__)
+
+
+class FeatureExternalResource(LifecycleModelMixin, models.Model):
+    class ResourceType(models.TextChoices):
+        # GitHub external resource types
+        GITHUB_ISSUE = "GITHUB_ISSUE", _("GitHub Issue")
+        GITHUB_PR = "GITHUB_PR", _("GitHub PR")
+
+    url = models.URLField()
+    type = models.CharField(max_length=20, choices=ResourceType.choices)
+
+    # JSON filed containing any metadata related to the external resource
+    metadata = models.TextField(null=True)
+    feature = models.ForeignKey(
+        Feature,
+        related_name="features",
+        on_delete=models.CASCADE,
+    )
+
+    class Meta:
+        ordering = ("id",)
+        constraints = [
+            models.UniqueConstraint(
+                fields=["feature", "url"], name="unique_feature_url_constraint"
+            )
+        ]
+
+    @hook(AFTER_SAVE)
+    def exectute_after_save_actions(self):
+        # Add a comment to GitHub Issue/PR when feature is linked to the GH external resource
```

Comment:
We need to add tests for this module

Reply:
I can't find a reference test for hooks, could you refer to some parts of the code?

Reply:
https://github.com/Flagsmith/flagsmith/blob/main/api/tests/unit/features/test_unit_features_models.py#L236

Reply:
I already tested this code in the create_feature_external_resources test and delete_feature_external_resources.